config/apply/ccsm/translate.js

```diff
@@ -435,53 +424,47 @@ export function translateResponse(customApiResponse, patientData) {
       conclusionCodes = mapResults(excisionResults, customExcisionCodes, standardHistologyCodes, conclusionCodes);
     }
 
-    if (diagnosticReportIndex !== -1) {
-      console.log('Found the diagnostic report reference');
-      // Update the DiagnosticReport in patientData with the results from the custom API
-      patientData[diagnosticReportIndex] = {
-        ...patientData[diagnosticReportIndex],
-        code: {
-          coding: [
-            ...patientData[diagnosticReportIndex].code.coding,
-            ...codings
-          ]
-        },
-        conclusionCode: conclusionCodes
-      };
+    let patient = patientData.find(pd => pd.resourceType === 'Patient');
+
+    // Create a DiagnosticReport resource from Order
+    let newDiagnosticReport = {
+      'resourceType': 'DiagnosticReport',
+      'id': order.OrderId,
+      'subject': { 'reference': 'Patient/' + patient.id },
+      'code': { 'coding': codings },
+      'conclusionCode': conclusionCodes,
+      'effectiveDateTime': order.ResultDate,
+      'status': 'final'
+    }
```

Comment:
Having a laboratory category code is required according to [U.S. Core](https://build.fhir.org/ig/HL7/US-Core/StructureDefinition-us-core-diagnosticreport-lab.html). This also aligns with the DiagnosticReport category codes in the [pilot test patients](https://upmchs.sharepoint.com/:f:/r/sites/CDCMitreUPMCCollaboration/Shared%20Documents/General/testing/Epic%20Results?csf=1&web=1&e=bfYf3v)

```suggestion
    let patient = patientData.find(pd => pd.resourceType === 'Patient');
    const laboratoryCategoryCode = {
      'coding': [
        {
          'system': 'http://terminology.hl7.org/CodeSystem/v2-0074',
          'code': 'LAB',
          'display': 'Laboratory'
        }
      ],
      'text': 'Laboratory'
    };

    // Create a DiagnosticReport resource from Order
    let newDiagnosticReport = {
      'resourceType': 'DiagnosticReport',
      'id': order.OrderId,
      'subject': { 'reference': 'Patient/' + patient.id },
      'category': [laboratoryCategoryCode],
      'code': { 'coding': codings },
      'conclusionCode': conclusionCodes,
      'effectiveDateTime': order.ResultDate,
      'status': 'final'
    }
```

Reply:
I don't think this is critical since we are generating a DiagnosticReport which can be processed by CQL. It does not quite matter if the resource is US Core compliant or not. But why not.

Reply:
I guess that's a good point, since we wouldn't ever be sending this anywhere besides our CQL engine. I believe we make some assumptions in parts of our CQL that things are aligned to U.S. Core, which is why I made a few comments about aligning the DiagnosticReport we create with https://build.fhir.org/ig/HL7/US-Core/StructureDefinition-us-core-diagnosticreport-lab.html. It's probably not critical, and for this case, definitely is not. I would say generally for the comments around aligning to U.S. Core, if it's a quick fix, it can't hurt so that the DR we're creating are as similar as possible to DRs we would see in the wild. But if adding certain elements becomes a larger lift, then we don't need to stress over it.

Reply:
`category` element was added in commit https://github.com/ccsm-cds-tools/ccsm-cds-service/pull/27/commits/02c87bed69d50ff406f2ea6f98e7224f3d6941a9. Thanks!